### ISSUE TITLE
help(sort): button to div change in 10.1.0 broke an accessibility check we had (WCAG 2.1 Criterion 2.5.5: Target Size)

### ISSUE THREAD

**Issue by bbarry** [angular/components#20105]
#### What are you trying to do?

I am not sure if this is really a help request or a if a documentation fix is needed or if there is an a11y issue to report. 

`@angular/material` `10.1.0` changed `mat-sort-header` from a `button` to a `div` in https://github.com/angular/components/commit/ea22e92a671ed9d64d72d7292fdd0dccb95559c8#diff-e5a01c54cda705d88178ecca63b2cb31

This change took the clickable height of the button on our implementation from 52px to 16px due to a global css rule we have for accessibility purposes:

```css
button {
  min-height: calc(1em + 2rem);
}
```

In my case the fix is easy enough and improves our code:

```css
button, div[role="button"] {
  min-height: calc(1em + 2rem);
}
```

I'm opening this issue because I am trying to figure out if one or more of the following should be done:

1. Document the button -> div change in the 10.1.0 release as a breaking change (Is this considered a breaking change? It clearly broke our automated tests).
2. Expose improved documentation about what is considered a breaking change.
3. Increase the minimum surface area for header buttons to meet the success criterion for WCAG 2.1 item ["2.5.5: Target Size"](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html).
4. Improve documentation related to accessibility features in general (eg: Material in general aims to meet WCAG 2.1 AAA? ) and specific to how table sorting is supposed to function.

**Comment by crisbeto** [comment]
I don't think the `button` -> `div` change is breaking since it's not a public API. As for the automated check failing, I'm not sure that it's a problem in practice because the clickable element isn't actually the `div[role="button"]`, but the `mat-sort-header` which is wrapping it and which has the same clickable area as before. The inner element needs the role, because that's where focus will land if the user is tabbing through.

**Comment by bbarry** [comment]
Yes the clickable area is the whole container but the cursor only updates for the text and we also have a hover underline added to the div tag here on hover.  We are attaching rules to the `button` (now `div[role="button"]`) because there historically were hover issues in older browsers we no longer support. I imagine that is why the pointer rule is on the button instead of the th as well.

**Comment by jelbourn** [comment]
As @crisbeto mentioned, we don't consider this a breaking change since the component's template are considered an internal implementation detail. Same goes for overriding the styles inside the components- it's pretty impossible for us to support all the arbitrary styles that someone might override on a component's internals.
